rasa/core/nlg/response.py

```diff
@@ -26,69 +26,6 @@ def __init__(self, responses: Dict[Text, List[Dict[Text, Any]]]) -> None:
         """
         self.responses = responses
 
-    def _matches_filled_slots(
-        self, filled_slots: Dict[Text, Any], response: Dict[Text, Any]
-    ) -> bool:
-        """Checks if the conditional response variation matches the filled slots."""
-        constraints = response.get(RESPONSE_CONDITION, [])
-        for constraint in constraints:
-            name = constraint["name"]
-            value = constraint["value"]
-            filled_slots_value = filled_slots.get(name)
-            if isinstance(filled_slots_value, str) and isinstance(value, str):
-                if filled_slots_value.casefold() != value.casefold():
-                    return False
-            elif filled_slots_value != value:
-                return False
-
-        return True
-
-    def _responses_for_utter_action(
-        self, utter_action: Text, output_channel: Text, filled_slots: Dict[Text, Any]
-    ) -> List[Dict[Text, Any]]:
-        """Returns array of responses that fit the channel, action and condition."""
-        default_responses = list(
-            filter(
-                lambda x: (x.get(RESPONSE_CONDITION) is None),
-                self.responses[utter_action],
-            )
-        )
-        conditional_responses = list(
-            filter(
-                lambda x: (
-                    x.get(RESPONSE_CONDITION)
-                    and self._matches_filled_slots(
-                        filled_slots=filled_slots, response=x
-                    )
-                ),
-                self.responses[utter_action],
-            )
-        )
```

Comment:
I see that you didn't write the clever code block! Its ok if you don't want to refactor it in `generator.py`. I hadn't seen this file until this and wasn't aware you've only re-arranged this block

Reply:
Yep, just added some context in [this reply](https://github.com/RasaHQ/rasa/pull/12575#discussion_r1246257045).
I think it would be really helpful if you could also look over the tests for the introduced functionality and think of any edge cases that I might have missed 🙏🏻

Reply:
Yes, I couldn't complete the review yesterday. I'm going to take a look at the rest of the files in the PR now